src/plugins/transport_topics/TransportTopics.qml

```diff
@@ -0,0 +1,49 @@
+import QtQuick 2.0
+import QtCharts 2.3
+import QtQuick.Controls 1.4
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Layouts 1.3
+
+TreeView {
+    width: 300
+    height: parent.height
+
+    id:tree
+    model: TopicsModel
+
+    TableViewColumn{title: "Topics"; role:"title" ;width: parent.width ;}
+
+    rowDelegate: Rectangle {
+        color: (styleData.selected)? "red" : (styleData.row % 2 == 0) ? "white" : "grey"
+        height: 50
+    }
+
+    itemDelegate: Item {
+        id :item
+        Text {
+            text: model.name
+            color: "black"
+        }
+    }
+
+
+    verticalScrollBarPolicy: Qt.ScrollBarAlwaysOff
+    horizontalScrollBarPolicy: Qt.ScrollBarAlwaysOff
+    headerVisible: true
+    onDoubleClicked:  { tree.subscribeTopic(); tree.expandCollapseFolder(); }
```

Comment:
I'm wondering if we really want this plugin to subscribe to topics. In the end, the plugin that will need the data will be the plotting plugin.

So maybe all this plugin should do is displaying all the topics / fields available, and making them draggable to any other plugins. Then the other plugin can decide what to do with it. If it needs the data, it will subscribe to it. Makes sense? This is how the old [TopicViewer](https://github.com/ignitionrobotics/ign-gui/blob/ign-gui0/include/ignition/gui/plugins/TopicViewer.hh) worked.

That plugin also displayed [this icon](https://github.com/ignitionrobotics/ign-gui/blob/ign-gui0/include/ignition/gui/images/graph_line.svg) next to each plottable field. Feel free to copy it here if you want :wink: